charts/argo-cd/templates/argocd-application-controller/deployment.yaml

```diff
@@ -117,29 +117,25 @@ spec:
       {{- with .Values.controller.extraContainers }}
         {{- toYaml . | nindent 6 }}
       {{- end }}
-    {{- if .Values.controller.nodeSelector }}
       nodeSelector:
-{{- toYaml .Values.controller.nodeSelector | nindent 8 }}
-    {{- end }}
-    {{- if .Values.controller.tolerations }}
+        node.coreweave.cloud/class: cpu
+
       tolerations:
-{{- toYaml .Values.controller.tolerations | nindent 8 }}
-    {{- end }}
-    {{- if .Values.controller.affinity }}
+        - key: is_cpu_compute 
+          operator: Exists 
       affinity:
-{{- toYaml .Values.controller.affinity | nindent 8 }}
-    {{- end }}
-    {{- with .Values.controller.topologySpreadConstraints }}
-      topologySpreadConstraints:
-      {{- range $constraint := . }}
-      - {{ toYaml $constraint | nindent 8 | trim }}
-        {{- if not $constraint.labelSelector }}
-        labelSelector:
-          matchLabels:
-            {{- include "argo-cd.selectorLabels" (dict "context" $ "name" $.Values.controller.name) | nindent 12 }}
-        {{- end }}
-      {{- end }}
-    {{- end }}
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: topology.kubernetes.io/region
+                    operator: In
+                    values:
+                      - ORD1
+                  - key: node.coreweave.cloud/class
+                    operator: In
+                    values:
+                      - cpu
```

Comment:
Mmmm, we can have these "defaults" in the _helpers, but definitely don't want to dispense with the configurability of the affinities, tolerations, etc.